src/rust/engine/fs/store/src/snapshot_ops.rs

```diff
@@ -366,6 +366,7 @@ impl IntermediateGlobbedFilesAndDirectories {
         RestrictedPathGlob::DirWildcard { wildcard, .. } => wildcard,
       };
 
+      #[allow(clippy::needless_collect)]
```

Comment:
Hm... both of these look like they could be immediately consumed without the collect (i.e., that the lint is correct). But we can fix that in a followup I suppose?

Reply:
The loop modifies the value being consumed to make the iterator, so we can't just remove the `collect`. We could potentially restructure this so that we do all of the modifications and collections inside a single iteration, but that would take some work :)

Reply:
FWIW this is https://github.com/rust-lang/rust-clippy/issues/6066

Reply:
I'll add a TODO comment linking to that issue.

Reply:
https://github.com/pantsbuild/pants/issues/12462